### DISCUSSION TITLE
New workspace and init ends early it seems

### DISCUSSION THREAD

**Discussion by mzupan** [terrakube-io/terrakube#1641] [General]
I'm running into an issue testing terrakube and when i make a new workspace and use opentofu when I try to do the default plan in the template I'm getting

```
***************************************
Initializing Terrakube Job 3 Step 35e40f52-2bec-4e83-bb9a-a53f241c2977
Running Tofu 1.8.7


***************************************
Running Tofu Init: 

Initializing the backend...

Successfully configured the backend "s3"! OpenTofu will automatically
use this backend unless the backend configuration changes.
Initializing modules...
Downloading git::https://github.com/org/tf-aws-modules?ref=main for base...
***************************************
Running Tofu PLAN
***************************************
���
��� Error: Module not installed
��� 
���   on main.tf line 1:
���    1: module "base" {
��� 
��� This module is not yet installed. Run "tofu init" to install all modules
��� required by this configuration.
���

```

It seems the init from terrakube just never finishes and it goes right into a plan

**Comment by alfespa17** [comment]
Are you using a private module? Maybe you can import the module using the terrakube registry and instead of using  ref main use an specific module version.

**Comment by mzupan** [comment]
coming back to this it seems to work with registry, so there is no way to make this work with git and modules outside the registry?

I have a lot of modules in a mono repo so i dont have 25 repos for ease of use

**Reply by alfespa17** [reply]
> coming back to this it seems to work with registry, so there is no way to make this work with git and modules outside the registry?
> 
> I have a lot of modules in a mono repo so i dont have 25 repos for ease of use

Hello @mzupan you can have múltiples modules in a mono repo setup, you can check this

https://github.com/AzBuilder/terrakube/issues/653

**Reply by alfespa17** [reply]
You can also have modules outside of the registry, I guess you are using some private git repository maybe using ssh key injection works for you so terrakube can authenticate and download the module. 

You can check

https://docs.terrakube.io/user-guide/vcs-providers/ssh#ssh-key-injection

**Reply by mzupan** [reply]
ahh thanks.. i was using `source = "git::https://github.com/` which i guess was actually using the PAT and not ssh key
